cmd/oomhero/main.go

```diff
@@ -106,6 +76,68 @@ func watchProcesses(ticks <-chan time.Time, getProcesses func() ([]proc.Process,
 	}
 }
 
+// reads warning and critical from environment or use the default ones.
+func readThresholdsFromEnvironment() {
+	warningEnv := envVarToUint64("WARNING", warning)
+	criticalEnv := envVarToUint64("CRITICAL", critical)
+
+	if warningEnv > 100 || criticalEnv > 100 {
+		log.Print("warning and critical must be lower or equal to 100")
+		return
+	} else if warningEnv > criticalEnv {
+		log.Print("warning must be lower or equal to critical")
+		return
+	}
+
+	warning = warningEnv
+	critical = criticalEnv
+}
+
+func readCooldownFromEnvironment() {
+	asString := os.Getenv("COOLDOWN")
+	if asString == "" {
+		return
+	}
+
+	val, err := time.ParseDuration(asString)
+	if err != nil {
+		log.Printf("error parsing COOLDOWN with time.ParseDuration: %v", err)
+		log.Print("falling back to legacy behavior")
+		legacyVal, err := strconv.ParseUint(asString, 10, 64)
+		if err != nil {
+			log.Printf("error parsing COOLDOWN as uint: %v", err)
+			return
+		}
+		log.Print("detected usage of deprecated format of COOLDOWN, migrate to time.ParseDuration format as soon as possible")
+		val = time.Duration(legacyVal) * time.Second
+	}
+
+	cooldownEnv := val
+
+	if cooldownEnv < 0*time.Second {
```

Comment:
I think this could also be `cooldownEnv < 0`.

Reply:
Changed to `cooldownEnv < 0`